qiskit/circuit/parameterexpression.py

```diff
@@ -246,6 +246,9 @@ def _apply_operation(
         self_expr = self._symbol_expr
 
         if isinstance(other, ParameterExpression):
+            if other._names is None:
+                other._names = {p.name: p for p in other._parameters}
+
```

Comment:
Can you add a comment explaining what's going on here? Or would this make more sense inside `_raise_if_parameter_names_conflict`, since that's where we populate `._names` for `self`?

Reply:
I simplified the logic here per @jakelishman's suggestion in: https://github.com/Qiskit/qiskit-terra/pull/7031/commits/28bdc39cf73c004d884019066d9ccc364e650749 so this is done implicitly now when we access `other._names`.